e2e/test_optimize_build.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# -*- indent-tabs-mode: nil; tab-width: 2; sh-indentation: 2; -*-
+
+set -x
+set -e
+set -o pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+OUTDIR="$(dirname "$SCRIPTDIR")/e2e-output"
+
+rm -rf "$OUTDIR"
+mkdir "$OUTDIR"
+
+tox -e e2e -n -r
+source .tox/e2e/bin/activate
+
+fromager \
+  --sdists-repo="$OUTDIR/sdists-repo" \
+  --wheels-repo="$OUTDIR/wheels-repo" \
+  --work-dir="$OUTDIR/work-dir" \
+  bootstrap 'stevedore==5.2.0' 2>&1 | tee "$OUTDIR/logfile.txt"
+
+pass=true
+
+if ! grep -q "Building source distribution" "$OUTDIR/logfile.txt"; then
+    echo "grep not successful"
```

Comment:
Better test failure message will help us debug if there are any failures in the future